### PR TITLE
[Rmc decouverte] Fix extractor

### PR DESCRIPTION
**Pull request by midas02**
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Extractor was no longer working. Fixed now.

### CODE REVIEW

**Comment by midas02** [comment]
Some comments.

1. The old code seemed to be overly complicated, but I've left it in place since I don't know if it still serves a purpose. The new code has been added with an If/else statement.
2. The code will fail if `brightcove_id` returns none. Is there a way of catching this?
3. The `data-video-id` key is referenced between ```<div class="brightcove_container">... </div>```. Should its detection be made conditional?

**Comment by dstftw** [comment]
2. Don't pass `default` and it won't be `None`.
3. Conditional to what?

**Comment by midas02** [comment]
2. That's not what I meant. Is there a way of returning an error message if the video_id is not found. Currently it just seems as if the programme has crashed.
3. Conditional to being caught in <div class="brightcove_container">.

**Comment by dstftw** [comment]
>a way of returning an error message if the video_id is not found

This is exactly what will happen if no `default` is passed.

>Conditional to being caught in `<div class="brightcove_container">`.

Explain **distinctly** what are you trying to do.

**Comment by midas02** [comment]
Well, a test to extract `data-video-id` only if it is being referenced in a class `"brightcove_container"`. I've tried to implement it, but in my opinion it will make the regex needlessly complicated and less robust.

File: youtube_dl/extractor/rmcdecouverte.py [hunk 1]
```diff
@@ -35,5 +35,8 @@ def _real_extract(self, url):
         video_id = self._match_id(url)
         webpage = self._download_webpage(url, video_id)
         brightcove_legacy_url = BrightcoveLegacyIE._extract_brightcove_url(webpage)
-        brightcove_id = compat_parse_qs(compat_urlparse.urlparse(brightcove_legacy_url).query)['@videoPlayer'][0]
+        if brightcove_legacy_url:
+            brightcove_id = compat_parse_qs(compat_urlparse.urlparse(brightcove_legacy_url).query)['@videoPlayer'][0]
+        else:
+            brightcove_id = self._html_search_regex(r'data-video-id="(.*?)"', webpage, 'error message')
```
Comment (dstftw) [inline youtube_dl/extractor/rmcdecouverte.py]: There is no point extracting empty video id. What the `error message`?

**Pull request closed without merging**